### PR TITLE
Mapper: allow 'expr' to be subclasses of ArrayOrNames

### LINKED ISSUE #236: `CachedMapper.__call__` should override types borrowed from `Mapper.__call__`

### PR DESCRIPTION
**Pull request by kaushikcfd**
Closes #236.

### CODE REVIEW

**Review by matthiasdiener — APPROVED** [review]
LGTM, thanks!

File: pytato/target/python/numpy_like.py [hunk 1]
```diff
@@ -549,8 +549,7 @@ def generate_numpy_like(expr: Union[Array, Mapping[str, Array], DictOfNamedArray
         numpy_backend=target.numpy_like_module_name_shorthand,
         numpy="np",
         vng=var_name_gen)
-    # type-ignore-reason: https://github.com/inducer/pytato/issues/236
```
Comment (inducer) [inline pytato/target/python/numpy_like.py:552]: Closes #236?

**Review by inducer — APPROVED** [review]
Thanks, LGTM!

**Comment by inducer** [comment]
(not merging yet to let you decide about the issue)

File: pytato/target/python/numpy_like.py (hunk 1 above)
Comment (kaushikcfd) [reply pytato/target/python/numpy_like.py:552]: Yes, I think it does.

Comment (kaushikcfd) [reply pytato/target/python/numpy_like.py:552]: Added it to the PR's description.

Comment (kaushikcfd) [reply pytato/target/python/numpy_like.py:552]: Thanks!

**Pull request merged**